pilot/cmd/pilot-discovery/main.go

```diff
@@ -120,9 +120,9 @@ func init() {
 			"It is recommended to be disable for highly available setups.")
 	discoveryCmd.PersistentFlags().StringVar(&serverArgs.Config.FileDir, "configDir", "",
 		"Directory to watch for updates to config yaml files. If specified, the files will be used as the source of config, rather than a CRD client.")
-	discoveryCmd.PersistentFlags().StringVarP(&serverArgs.Config.ControllerOptions.WatchedNamespace, "appNamespace",
+	discoveryCmd.PersistentFlags().StringVarP(&serverArgs.Config.ControllerOptions.WatchedNamespaces, "appNamespace",
```

Comment:
I think this should be added to values.yaml and/or mesh config ( there are discussions of merging them so it'll be the same thing )

Reply:
@costinm We've been looking at something similar for our operator and decided to split the namespaces out into a separate custom resource, the reasons for doing this were

- being able to enforce different permissions for modifying the set of namespaces
- having the operator validate the configuration, especially ensuring the namespace sets remain distinct (soft multi-tenancy)
- provide information for the operator to update roles to access the resources in those namespaces (things like nodes are still a problem for us)
- allow the operator to configure the networking (multi-tenant/network policy) on behalf of the namespaces

Do you think it's worth doing something similar in the community?